### PR TITLE
Add Medusa to E-commerce

### PR DESCRIPTION
**Pull request by SimonFischerSE**
Thank you for taking the time to work on a PR for Awesome-Selfhosted!

To ensure your PR is dealt with swiftly please check the following:

- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] Format your submission as follows, where `Demo` and `Clients` are optional.
  Do not add a duplicate `Source code` link if it is the same as the main link.
  Keep the short description under 250 characters and use [sentence case](https://en.wikipedia.org/wiki/Letter_case#Sentence_case)
  for it, even if the project's webpage or readme uses another capitalisation
  such as title case, all caps, small caps or all lowercase.
  `Demo` links should only be used for interactive demos, i.e. not video demonstrations.
  ``- [Name](http://homepage/) - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that depend on proprietary services outside the user's control must be marked `⚠`.
  ``- [Name](http://homepage/) `⚠` - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `License` `Language` ``
- [x] Additions that are not [Free software](https://en.wikipedia.org/wiki/Free_software)
  must be added to `non-free.md` and marked `⊘ Proprietary`:
  ``- [Name](http://homepage/) `⊘ Proprietary` - Short description, under 250 characters, sentence case. ([Demo](http://url.to/demo), [Source Code](http://url.of/source/code), [Clients](https://url.to/list/of/related/clients-or-apps)) `Language` ``
- [x] Additions are inserted preserving alphabetical order.
- [x] Additions are not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [awesome-analytics](https://github.com/onurakpolat/awesome-analytics), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/).
- [x] The `Language` tag is the main **server-side** requirement for the software. Don't include frameworks or specific dialects.
- [x] Any license you add is in our [list of licenses](https://github.com/awesome-selfhosted/awesome-selfhosted/blob/master/README.md#list-of-licenses).
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted/pulls), including closed ones.
- [x] Any category you are creating has the minimum requirement of 3 items.
  If not, your addition may be inserted into `Misc/Other`.
- [x] Any software project you are adding to the list is actively maintained.
- [x] The pull request title is informative, unlike "Update README.md".
  Suggested titles: "Add aaa to bbb" for adding software aaa to section bbb,
  "Remove aaa from bbb" for removing, "Fix license for aaa", etc.

### CODE REVIEW

**Comment by alexanderl19** [comment]
I saw that Medusa previously got rejected in #2711 for not being an end-user application. I'm not quite sure this pushes it over the edge, but they do have [Next.js](https://github.com/medusajs/nextjs-starter-medusa) and [Gatsby](https://github.com/medusajs/gatsby-starter-medusa) templates.

**Comment by nodiscc** [comment]
> the whole setup process is pretty simple (run CLI tool, edit config, deploy).

If that it the case it would qualify. If you could point me to example deployment instructions I could take another look.

**Comment by SimonFischerSE** [comment]
> 

You can find guides on deployment here https://docs.medusajs.com/how-to/deploying-on-heroku

**Comment by nodiscc** [comment]
> deploying-on-heroku

How do I deploy this on a server under my own control and not on a proprietary PaaS?

**Comment by alexanderl19** [comment]
The Medusa starter [1] is a Node.js application, so it can be deployed to your own hardware with a variety of methods, like PM2 or in a Docker container. Medusa doesn't force you to use a cloud provider, and has pretty good support if you want to avoid it entirely. For example, if you prefer not to use Amazon S3, you can host your own MinIO server [2].

Reaction Commerce, another project on this list [3], seems to have a pretty similar deployment process and project structure. It also has three components (API, Admin Panel, Storefront), and a template repo [4] for configuration and setup.

[1]: Running `medusa new my-medusa-store` creates https://github.com/medusajs/medusa-starter-default
[2]: https://docs.medusajs.com/add-plugins/minio
[3]: https://github.com/awesome-selfhosted/awesome-selfhosted/blob/3e61ab40f5a76f599b9b532b62dfd58ed67d4b9d/README.md?plain=1#L776
[4]: https://github.com/reactioncommerce/reaction

**Review by nodiscc — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -760,6 +760,7 @@ _Related: [Community-Supported Agriculture (CSA)](#community-supported-agricultu
 - [CoreShop](https://www.coreshop.org) - CoreShop is a e-commerce plugin for Pimcore. ([Source Code](https://github.com/coreshop/CoreShop)) `GPL-3.0` `PHP`
 - [Drupal Commerce](https://drupalcommerce.org) - Drupal Commerce is a popular e-commerce module for Drupal CMS, with support for dozens of payment, shipping, and shopping related modules. ([Source Code](https://github.com/drupalcommerce/commerce)) `GPL-2.0` `PHP`
 - [Magento](https://magento.com/) - Leading provider of open omnichannel innovation. ([Demo](https://magento.com/schedule-a-demo), [Source Code](https://github.com/magento/magento2)) `OSL-3.0` `PHP`
+- [Medusa](https://medusajs.com/) - Medusa is an open-source headless commerce engine that enables developers to create amazing digital commerce experiences ([Demo](https://demo.medusajs.com/a/orders?offset=0&limit=15), [Source Code](https://github.com/medusajs/medusa), `MIT` `JavaScript`
```
Comment (nodiscc) [inline README.md:763]: @SimonFischerSE please change the language tag to `Nodejs`

> The Language tag is the main server-side requirement for the software
> https://github.com/medusajs/medusa-starter-default#setting-up-your-store
> npm install

**Review by nodiscc — CHANGES_REQUESTED** [review]

**Review by nodiscc — APPROVED** [review]

**Pull request merged**